temporal-testing/src/main/java/io/temporal/internal/testservice/TestServiceServer.java

```diff
@@ -0,0 +1,30 @@
+/*
+ *  Copyright (C) 2020 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package io.temporal.internal.testservice;
+
+public class TestServiceServer {
+
+  public static void main(String[] args) throws InterruptedException {
+    TestWorkflowService service = new TestWorkflowService(7233);
+    if (service != null) {
```

Comment:
I don't understand this check and the shutdown part. I think it should call shutdown on a shutdown hook.

Reply:
Also it should never ever be `null` after we've just initialized it.